connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -220,6 +220,8 @@ public void stop() {
 
         workerMetricsGroup.close();
         connectorStatusMetricsGroup.close();
+
+        workerConfigTransformer.close();
```

Comment:
Looking at the initialization of `workerConfigTransformer` I see it should be made final. 

And then I notice that this is the case for `connectorClientConfigOverridePolicy` and all the class members of `ConnectorStatusMetricsGroup`. @tombentley do you mind tightening these types as well?